src/Adapter/Product/AbstractProductHandler.php

```diff
@@ -129,4 +130,29 @@ protected function validateField(Product $product, string $field, int $errorCode
             );
         }
     }
+
+    /**
+     * @param Product $product
+     * @param int $errorCode
+     *
+     * @throws CannotUpdateProductException
+     * @throws ProductException
+     */
+    protected function performUpdate(Product $product, int $errorCode): void
+    {
+        try {
+            if (false === $product->update()) {
+                throw new CannotUpdateProductException(
+                    sprintf('Failed to update product #%s', $product->id),
+                    $errorCode
+                );
+            }
+        } catch (PrestaShopException $e) {
+            throw new ProductException(
+                sprintf('Error occurred when trying to update product #%s', $product->id),
```

Comment:
```suggestion
                sprintf('Error occurred when trying to update product #%d', $product->id),
```

Reply:
what difference does it make? :thinking:

Reply:
We must use `%f`, `%s`, `%d` depending on the type otherwise the sprint has no aim and a simple concatenation is enough :sweat_smile: